source/renderer/app/components/wallet/mnemonic-input/MnemonicAutocompleteContainer.tsx

```diff
@@ -0,0 +1,169 @@
+import React, {
+  ClipboardEventHandler,
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { startsWith } from 'lodash/fp';
+import { GlobalListeners } from 'react-polymorph/lib/components/HOC/GlobalListeners';
+import { MnemonicAutocompleteLayout } from './MnemonicAutocompleteLayout';
+
+interface MnemonicAutocompleteContainerProps {
+  onChange: (value: string) => void;
+  onConfirmSelection: () => void;
+  onPaste: ClipboardEventHandler<HTMLInputElement>;
+  ordinalNumber: number;
+  value: string;
+  options: string[];
+  maxVisibleOptions: number;
+  disabled: boolean;
+  error: boolean;
+  reset: boolean;
+  noResultsMessage: string;
+  inputRef: RefObject<HTMLInputElement>;
+}
+
+const MnemonicAutocompleteContainer = ({
+  onChange,
+  onConfirmSelection,
+  onPaste,
+  reset,
+  ordinalNumber,
+  value,
+  options,
+  maxVisibleOptions,
+  disabled,
+  error,
+  noResultsMessage,
+  inputRef,
+}: MnemonicAutocompleteContainerProps) => {
+  const initialState = useMemo(
```

Comment:
I think useMemo is not necessary as you can have the same result using [lazy-initial-state](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state)

Reply:
Never mind, I see it is used again on reset.